fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -189,11 +200,33 @@ public Response head() {
 
             // we set the content-type explicitly to avoid content-negotiation from getting in the way
             builder.type(mediaType.toString());
+        } else {
+            final MediaType mediaType = RDFMediaType.TURTLE_TYPE;
+            final RdfStream stream = getResourceAsStream();
```

Comment:
I like the other changes in this PR, but I'm worried about the performance implications of this, esp. for resources with many links to other repository children.  Producing the output in order to get its size seems very counter-productive.  What is the point of doing a HEAD request if it isn't fast?